### DISCUSSION TITLE
Exporting comments

### DISCUSSION THREAD

**Discussion by romange** [cue-lang/cue#1178] [Q&A] [answered]
Sometimes one needs to generate text without applying any transformations. 
For example, cloud-init yml file must have `#cloud-config` shebang at the beginning. Is it possible to specify such text blob within cue file so it would be preserved during the export?

Specifically, I would like to generate the following yaml file

```
#cloud-config
package_update: true
package_upgrade: true
groups:
- docker
```

It can be trivially achieved by the wrapping bash script but I wonder if this is something cuelang would like to support natively - preserving non-functional annotations that are interpreted by 3rd party systems that execute these configuration scripts.

**Comment by verdverm** [comment]
Maybe you are looking for `"""` (triple quote strings)? https://cuelang.org/docs/tutorials/tour/types/stringlit/ See also the "raw" strings on the next page.

You can also have `"#cloud-config": <...>` as a label, which is not a definition. Note the double quotes around the label.

If you are looking for something specific to outputting that first line, an example of the desired input / output would be helpful.

**Reply by romange** [reply]
Thanks! I updated the original question. It's most surely not related to raw strings or triple quotes.  
I also do not think labels can help here.

**Reply by verdverm** [reply]
ah, so the issue appears to be that the export process does not bring the comments through.

If however, you add `, cue.Docs(true)` to this line https://github.com/cue-lang/cue/blob/master/pkg/encoding/yaml/manual.go#L38

You will get the following, which I suspect would resolve your issue

```
$ cat test.cue 
//cloud-config
foo: "bar"

hello: {
        // occupy
        world: "from mars?"
}

$ cue export --out yaml test.cue 
# cloud-config
foo: bar
hello:
  # occupy
  world: from mars?
  ```
  
  although the spacing might be off...

**Comment by myitcv** [comment, marked as answer]
@romange as @verdverm alluded to above, the `cue export` command does not (currently) have a way of saying "please also export comments". Neither does a `cue cmd`-defined command either, nor the Go API. This feels like a reasonable option (likely amongst others) we could support. Please feel free to open a [feature request](https://github.com/cue-lang/cue/issues/new/choose).

**Reply by romange** [reply]
https://github.com/cue-lang/cue/issues/1180

**Reply by myitcv** [reply]
Thank you!
